source/solvers/multiphysics_interface.cc

```diff
@@ -0,0 +1,26 @@
+#include <solvers/multiphysics_interface.h>
+
+#include "solvers/heat_transfer.h"
+
+
+template <int dim>
+MultiphysicsInterface<dim>::MultiphysicsInterface(
+  const SimulationParameters<dim> &                            nsparam,
+  std::shared_ptr<parallel::DistributedTriangulationBase<dim>> p_triangulation,
+  std::shared_ptr<SimulationControl> p_simulation_control)
+  : multiphysics_parameters(nsparam.multiphysics)
+{
+  if (multiphysics_parameters.fluid_dynamics)
+    {
+      active_physics.push_back(PhysicsID::fluid_dynamics);
+    }
+  if (multiphysics_parameters.heat_transfer)
+    {
+      active_physics.push_back(PhysicsID::heat_transfer);
+      physics[PhysicsID::heat_transfer] = std::make_shared<HeatTransfer<dim>>(
```

Comment:
I didn't know this make_shared method! Interesting

Reply:
So std::shared_ptr are like regular pointers, except that when they go out of scope they are automatically deallocated. They are about just as fast, but very safe from a memory point of view. Prevent leaks and etc.